# Epitope Table 19-AA Held-Out Preparation

This notebook prepares the new IEDB epitope table export for the same 19-amino-acid held-out idea used in the HLA runs. Rows are kept only when the epitope is a linear, unmodified, canonical 9-mer using the 20 standard amino acids. Any peptide name with non-canonical characters such as `+`, `X`, `B`, `Z`, `J`, brackets, or other uncertainty/modification notation is excluded. Rows with populated `Modified Residue(s)` or `Modifications` fields are also excluded before deduplication.

**Important limitation:** this XLSX is an epitope catalogue. It does not include binding labels or HLA allele fields, so AUC, confusion matrices, and the AE/one-hot/BLOSUM supervised comparison cannot be computed from this file alone.

In [1]:
# Recreate all generated CSV/SVG/TXT outputs from the workbook.
%run epitope_table_heldout_run.py --no-notebook


Workbook: epitope_table_export_1779969300.xlsx
Raw epitope rows: 9833
Canonical retained rows before dedupe: 9648
Unique canonical 9-mers: 9648
Default held-out amino acid: W
Training/CV peptides without W: 8577
Evaluation peptides with W: 1071
Supervised model performance was not computed because this XLSX has no binding labels or HLA allele column.


## Current Saved Summary

| Metric | Value |
| --- | --- |
| Raw epitope rows | 9833 |
| Retained canonical 9-mer rows before dedupe | 9648 |
| Unique canonical 9-mers | 9648 |
| Train/CV peptides without W | 8577 |
| Evaluation peptides with W | 1071 |
| Consensus, all canonical 9-mers | RLYPPLLLK |
| Consensus, no W train/CV | RLYPPLLLK |
| Consensus, with W evaluation | RLWWWWWWK |
| Excluded: modified_or_uncertain_annotation | 185 |

## Held-Out Amino-Acid Candidates

The default split is held-out `W` so this epitope-table preparation stays comparable with the earlier HLA W-heldout runs. The candidate table below also shows that the workbook can support other held-out choices; set `DEFAULT_HELD_OUT_AA` in `epitope_table_heldout_run.py` and rerun if you want a different residue.

| Held_Out_AA | Train_Count_No_AA | Evaluation_Count_With_AA | Anchor_P2_or_P9_Count | Anchor_Only_P2_or_P9_Count | Rank_By_Train_Count | Rank_By_Evaluation_Count |
| --- | --- | --- | --- | --- | --- | --- |
| C | 8965 | 683 | 67 | 57 | 1 | 20 |
| W | 8577 | 1071 | 134 | 102 | 2 | 19 |
| M | 7734 | 1914 | 526 | 418 | 3 | 18 |
| H | 7705 | 1943 | 457 | 366 | 4 | 17 |
| D | 7616 | 2032 | 55 | 44 | 5 | 16 |
| E | 7258 | 2390 | 202 | 138 | 6 | 15 |
| N | 7242 | 2406 | 69 | 52 | 7 | 14 |
| Q | 7211 | 2437 | 406 | 286 | 8 | 13 |
| P | 6289 | 3359 | 326 | 215 | 9 | 12 |
| G | 6210 | 3438 | 178 | 91 | 10 | 11 |
| F | 6194 | 3454 | 545 | 349 | 11 | 10 |
| Y | 5943 | 3705 | 1539 | 1066 | 12 | 9 |
| I | 5847 | 3801 | 1136 | 739 | 13 | 8 |
| R | 5523 | 4125 | 1284 | 731 | 14 | 7 |
| T | 5470 | 4178 | 1330 | 835 | 15 | 6 |
| A | 5356 | 4292 | 543 | 274 | 16 | 5 |
| V | 5112 | 4536 | 1809 | 1123 | 17 | 4 |
| S | 5085 | 4563 | 793 | 412 | 18 | 3 |
| K | 3428 | 6220 | 4586 | 2822 | 19 | 2 |
| L | 3197 | 6451 | 3044 | 1442 | 20 | 1 |

## Default W Split Preview

Training/CV peptides contain no `W` anywhere. Evaluation peptides contain `W` at one or more positions.

Train/CV preview:

| Peptide | Fold | Held_Out_AA | Held_Out_Positions | Source_Organism | Species |
| --- | --- | --- | --- | --- | --- |
| AAAAALQAK | 1 | W |  | Homo sapiens | Homo sapiens |
| AAAACTTMK | 2 | W |  | Homo sapiens | Homo sapiens |
| AAAKPTPPK | 3 | W |  | Homo sapiens | Homo sapiens |
| AAAKTLNPK | 4 | W |  | unidentified | unidentified |
| AAAKVILKK | 5 | W |  | Homo sapiens | Homo sapiens |
| AAAPAATAK | 1 | W |  | unidentified | unidentified |
| AAAPALQGK | 2 | W |  | unidentified | unidentified |
| AAAPPLSFR | 3 | W |  | unidentified | unidentified |

Evaluation preview:

| Peptide | Held_Out_AA | Held_Out_Positions | Source_Organism | Species |
| --- | --- | --- | --- | --- |
| AALQSAWQG | W | 7 | Mycobacterium tuberculosis H37Rv | Mycobacterium tuberculosis |
| AANTVIWDY | W | 7 | SARS coronavirus Tor2 | SARS-CoV1 |
| AAYAWPFYK | W | 5 | Homo sapiens | Homo sapiens |
| ADFKLFFRW | W | 9 | Escherichia coli | Escherichia coli |
| AEFWDVFLS | W | 4 | Giardia lamblia ATCC 50803 | Giardia duodenalis |
| AEHDPWWAV | W | 6;7 | Corynebacterium glutamicum | Corynebacterium glutamicum |
| AELIDSFTW | W | 9 | Campylobacter jejuni subsp. jejuni NCTC 11168 = ATCC 700819 | Campylobacter jejuni |
| AEMRETHWL | W | 8 | Shigella dysenteriae | Shigella dysenteriae |

## Sequence-Logo Style Composition

The files below are information-content stack plots generated from position-wise amino-acid frequencies. The underlying frequency tables are saved as CSVs in `epitope_table_new_data_results/`.

All canonical 9-mers:

![All canonical 9-mer logo](epitope_table_new_data_results/Epitope_Table_Logo_All.svg)

Train/CV peptides with no `W`:

![Train logo](epitope_table_new_data_results/Epitope_Table_Logo_Train_No_W.svg)

Evaluation peptides with `W`:

![Evaluation logo](epitope_table_new_data_results/Epitope_Table_Logo_Evaluation_With_W.svg)

## Downstream Modeling Note

To run the full binding-prediction comparison later, join these canonical peptides to a labeled assay/HLA table that has at least `peptide`, `label`, and `allele`. After that join, the existing HLA workflow can be reused: keep one allele, train five folds on peptides without the held-out amino acid, and evaluate ensemble predictions on peptides containing the held-out amino acid.